upload/views/base.py

```diff
@@ -10,7 +11,19 @@
 log = logging.getLogger(__name__)
 
 
-class GetterMixin:
+class ShelterMixin:
+    def is_shelter_request(self) -> bool:
+        """
+        Returns true when the incoming request originated from a Shelter.
+        Shelter adds an `X-Shelter-Token` header which contains a shared secret.
+        Use of that shared secret allows certain priviledged functionality that normal
+        uploads cannot access.
+        """
+        shelter_token = self.request.META.get("HTTP_X_SHELTER_TOKEN")
+        return shelter_token and shelter_token == settings.SHELTER_SHARED_SECRET
```

Comment:
Makes sense. Is this a go-to approach when trying to identify specific requests? (for my own learning heh)

Reply:
Makes sense, I recall doing some identification through headers, cool!